controller/hybridgateway/builder/kongplugin.go

```diff
@@ -223,3 +239,87 @@ func translateResponseModifier(filter gwtypes.HTTPRouteFilter) (transformerData,
 	}
 	return plugin, err
 }
+
+type requestRedirectConfig struct {
+	KeepIncomingPath bool   `json:"keep_incoming_path"`
+	Location         string `json:"location"`
+	StatusCode       int    `json:"status_code"`
+}
+
+func translateRequestRedirect(filter gwtypes.HTTPRouteFilter) (requestRedirectConfig, error) {
+	rr := filter.RequestRedirect
+
+	if rr == nil {
+		return requestRedirectConfig{}, errors.New("RequestRedirect filter config is missing")
+	}
+
+	// GWAPI default status code is 302
+	plugin := requestRedirectConfig{StatusCode: 302}
+
+	if rr.StatusCode != nil {
+		plugin.StatusCode = *rr.StatusCode
+	}
+
+	locHost := translateRequestRedirectHostname(rr)
+	locPath, err := translateRequestRedirectPath(rr)
+	if err != nil {
+		return requestRedirectConfig{}, err
+	}
+	if locPath == "" {
+		plugin.KeepIncomingPath = true
+		locPath = "/"
+	}
+
+	plugin.Location = locHost + locPath
+	return plugin, nil
+}
+
+func translateRequestRedirectHostname(rr *gatewayv1.HTTPRequestRedirectFilter) string {
+	if rr.Hostname == nil {
+		return ""
+	}
+
+	// when no scheme specified we assume `http` (as KIC does) but we should preserve the actual scheme instead.
+	// this cannot be done with direct filter -> kong plugin conversion.
+	// See https://github.com/Kong/kong-operator/issues/2466.
+	host := lo.FromPtrOr(rr.Scheme, "http") + "://"
+	host += lo.FromPtrOr((*string)(rr.Hostname), "")
```

Comment:
```suggestion
	host += string(lo.FromPtrOr((rr.Hostname), ""))
```
WDYT? It seems easier to understand